src/Microsoft.AspNetCore.SignalR.Core/Internal/HubMethodDescriptor.cs

```diff
@@ -34,6 +32,25 @@ public HubMethodDescriptor(ObjectMethodExecutor methodExecutor, IEnumerable<IAut
                 IsChannel = true;
                 StreamReturnType = channelItemType;
             }
+
+            // Take out synthetic arguments that will be provided by the server, this list will be given to the protocol parsers
+            ParameterTypes = methodExecutor.MethodParameters.Where(p =>
+            {
+                // Only streams can take CancellationTokens currently
+                if (IsStreamable && p.ParameterType == typeof(CancellationToken))
```

Comment:
Can we put this check into the `SyntheticArgumentManager`? Something like `SyntheticArgumentManager.IsSyntheticArgumentType(...)`

Reply:
Disregard this comment now.